### PR TITLE
Added multiple vite templates with static site building.

### PR DESCRIPTION
**Pull request by StringentDev**
Added the following Vite variants:
 - Vanilla
 - Vue 3
 - Preact
 - React
 - Lit
 - Svelte
 - Typescript variants of those listed

Hopefully this gets merged to make it just that tad bit easier to get started with Vite and Railway. They also include the getting started buttons.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (JakeCooper) [inline yarn.lock:4]: Can you remove the yarn.lock here?

**Review by JakeCooper — APPROVED** [review]
Approved pending the yarn.lock removal!

File: yarn.lock (hunk 1 above)
Comment (StringentDev) [reply yarn.lock:4]: Sure

**Comment by StringentDev** [comment]
@JakeCooper Ok done? Is there the green light?

**Pull request merged**